Lorem Ipsum XR Translator/Assets/Scripts/Behaviors/SceneAnalyzer.cs

```diff
@@ -0,0 +1,226 @@
+    void Start()
+    {
+        CaptionController = GetComponent<CaptionController>();
+    }
+
+    /// <summary>
+    /// The entry point for starting an analysis. This is called from the scene to begin taking a picture
+    /// </summary>
+    /// Camera code largely sourced from: https://learn.microsoft.com/en-us/windows/mixed-reality/develop/unity/locatable-camera-in-unity
+    /// There might be an issue with camera capture occurring on main thread, and not asynchronously at all.
+    public void StartCapture()
+    {
+        DebugText.text = "Beginning screenshot process";
+        PhotoCapture.CreateAsync(false, OnPhotoCaptureCreated);
+    }
+
+    /// <summary>
+    /// Initialize our photo capture object
+    /// </summary>
+    /// <param name="captureObject">The photo capture data</param>
+    private void OnPhotoCaptureCreated(PhotoCapture captureObject)
+    {
+        DebugText.text = "Photo Created";
+        photoCaptureObject = captureObject;
+
+        Resolution cameraResolution = PhotoCapture.SupportedResolutions.OrderByDescending((res) => res.width * res.height).First();
+
+        CameraParameters c = new CameraParameters();
+        c.hologramOpacity = 0.0f;
+        c.cameraResolutionWidth = cameraResolution.width;
+        c.cameraResolutionHeight = cameraResolution.height;
+        c.pixelFormat = CapturePixelFormat.BGRA32;
+
+        captureObject.StartPhotoModeAsync(c, OnPhotoModeStarted);
+
+    }
+    /// <summary>
+    /// Handle photo capture
+    /// </summary>
+    /// <param name="result">Whether the capture initialized successfully</param>
+    private void OnPhotoModeStarted(PhotoCapture.PhotoCaptureResult result)
+    {
+        if (result.success)
+        {
+            photoCaptureObject.TakePhotoAsync(OnCapturedPhotoToMemory);
+        }
+        else
+        {
+            DebugText.text = "Unable to start photo mode!";
+        }
+    }
+
+    /// <summary>
+    /// Function that handles what to do with captured photo data
+    /// </summary>
+    /// <param name="result">The result of the photo capture operation</param>
+    /// <param name="photoCaptureFrame">The captured photo data</param>
+    private void OnCapturedPhotoToMemory(PhotoCapture.PhotoCaptureResult result, PhotoCaptureFrame photoCaptureFrame)
+    {
+        if (result.success)
+        {
+
+            // Create our Texture2D for use and set the correct resolution
+            Resolution cameraResolution = PhotoCapture.SupportedResolutions.OrderByDescending((res) => res.width * res.height).First();
+            Texture2D targetTexture = new Texture2D(cameraResolution.width, cameraResolution.height);
+            // Copy the raw image data into our target texture
+            photoCaptureFrame.UploadImageDataToTexture(targetTexture);
+            this.image = targetTexture;
+            if (DebugQuad)
+            {
+                DebugQuad.GetComponent<Renderer>().material.mainTexture = targetTexture;
+            }
```

Comment:
Is it a temporal thing or part of the features?

Reply:
I don't quite understand what this does😅

Reply:
This one is just for testing. This is what renders the captured screenshot to the quad in the player's face so that we can see the result of the screen capture. We've proven that this works in our last test, so it can be safely removed in the future, but I'll leave it for now (since it at least has safe handling and doesn't run if no debugquad is found).